x-pack/plugins/security_solution/cypress/e2e/detection_rules/related_integrations.cy.ts

```diff
@@ -117,19 +126,24 @@ describe('Related integrations', () => {
       };
 
       before(() => {
-        cleanPackages();
-        installAwsCloudFrontWithPolicy();
+        cleanFleet().then(() => {
+          installAwsCloudFrontWithPolicy();
+        });
+      });
+
+      beforeEach(() => {
         visit(DETECTIONS_RULE_MANAGEMENT_URL);
```

Comment:
I understand the change done here in order to achieve less flakiness, the only downside is the execution is going to be slower. This change has inspired me for a future refactor to try to avoid the mentioned downside at the same time we maintain the reliability, so, thank you!

Reply:
> the only downside is the execution is going to be slower

Yeah but not much. Stability is more important than speed. The speed is still acceptable locally (at least the page reload doesn't make it _way slower_). On the CI we can parallelize.

> This change has inspired me for a future refactor to try to avoid the mentioned downside at the same time we maintain the reliability, so, thank you!

🙌

Reply:
@MadameSheema Btw, mainly the flake was caused by the way you implemented the Fleet packages cleanup logic. Test failure could lead to a broken state of the Fleet saved objects where:

- a package couldn't be deleted because it was referenced from a package policy
- this package policy couldn't be deleted either because its parent agent policy was deleted by the test at a previous stage

Fleet is hard 🙂

Reply:
True story, sorry for adding flakiness  🫣

Reply:
@MadameSheema you shouldn't be sorry! Fleet's API is not really documented. There's a [Swagger spec](https://github.com/elastic/kibana/blob/8.2/x-pack/plugins/fleet/common/openapi/README.md) that doesn't have any comments for humans, and [this doc](https://docs.elastic.dev/kibana-dev-docs/api/fleet) doesn't help much. There's only [this doc](https://github.com/elastic/kibana/blob/a547d4a087950dff559e6a2fd3ee7eb8ee1f4a2a/x-pack/plugins/fleet/dev_docs/data_model.md) that sheds some light on what saved objects Fleet uses under the hood. So we will all continue learning Fleet stuff together: what entities do they have, what are dependencies between them, how to use the API, etc etc.